doc/howto/multi_level_argument_parser/multi_level_arg_parse.cpp

```diff
@@ -0,0 +1,97 @@
+#include <seqan3/argument_parser/all.hpp>
+
+using namespace seqan3;
+
+// =====================================================================================================================
+// pull
+// =====================================================================================================================
+
+struct pull_arguments
+{
+    std::string repository{};
+    std::string branch{};
+    bool progress{false};
+};
+
+int run_git_pull(argument_parser & parser)
+{
+    pull_arguments args{};
+
+    parser.add_positional_option(args.repository, "The repository name to pull from.");
+    parser.add_positional_option(args.branch, "The branch name to pull from.");
+
+    try
+    {
+        parser.parse();
+    }
+    catch (parser_invalid_argument const & ext)
+    {
+        debug_stream << "[Error] " << ext.what() << "\n";
+        return -1;
+    }
+
+    return 0;
+}
+
+// =====================================================================================================================
+// push
+// =====================================================================================================================
+
+struct push_arguments
+{
+    std::string repository{};
+    std::vector<std::string> branches{};
+    bool push_all{false};
+};
+
+int run_git_push(argument_parser & parser)
+{
+    push_arguments args{};
+
+    parser.add_positional_option(args.repository, "The repository name to push to.");
+    parser.add_positional_option(args.branches, "The branch names to push (if none are given, push current).");
+
+    try
+    {
+        parser.parse();
+    }
+    catch (parser_invalid_argument const & ext)
+    {
+        debug_stream << "[Error] " << ext.what() << "\n";
+        return -1;
+    }
+
+    return 0;
+}
+
```

Comment:
```suggestion
// =====================================================================================================================
// main
// =====================================================================================================================
```

If you use these in the code, add one before main